File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [0];
    let rest = []
    for (let i = 0; i < speeds.length; i++){
        rest.push(Math.ceil((100 - progresses[i]) / speeds[i]))
    }
    
    let max = rest[0];

    for (let i = 0, j = 0; i< rest.length; i++){
        if(rest[i] <= max) answer[j] += 1;
        else {
            max = rest[i];
            answer[++j] = 1;
        }
    }

    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = 0
    d = 0
    
    while len(progresses)> 0:
        if (progresses[0] + day * speeds[0]) >= 100:
            progresses.pop(0)
            speeds.pop(0)
            d += 1
        else:
            if d > 0:
                answer.append(d)
                d = 0
            day += 1
            
    answer.append(d)
    
    return answer